### PR TITLE
Add mass mailing keep archives

### PR DESCRIPTION
**Pull request by nicomacr**
This module adds an option on mass mailing campaings to allow you set if you
 +want to keep archives (mails sent) or not.

### CODE REVIEW

**Comment by nicomacr** [comment]
moved https://github.com/OCA/crm/pull/66

**Comment by rafaelbn** [comment]
Thanks. Please review https://github.com/OCA/maintainer-tools, you miss https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: mass_mailing_keep_archives/mass_mailing.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        """ Override method to add keep_archives functionalty """
+        res = super(MailComposeMessage, self).get_mail_values(
+            wizard, res_ids)
+        # use only for allowed models in mass mailing
+        if wizard.composition_mode == 'mass_mail' and \
+                (wizard.mass_mailing_name or wizard.mass_mailing_id) and \
+                wizard.model in [item[0] for item in self.env[
+                    'mail.mass_mailing']._get_mailing_model()]:
+            mass_mailing = wizard.mass_mailing_id
+            # opcion solo desde mass mailing, no para composiciones que creen
```
Comment (hbrunn) [inline mass_mailing_keep_archives/mass_mailing.py]: write English comments

File: mass_mailing_keep_archives/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (hbrunn) [inline mass_mailing_keep_archives/__openerp__.py:5]: use short license headers: https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L1

File: mass_mailing_keep_archives/mass_mailing.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (hbrunn) [inline mass_mailing_keep_archives/mass_mailing.py]: have ome file per model in /models: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#directories

File: mass_mailing_keep_archives/mass_mailing.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from openerp import models, api, fields
+
+
+class MassMailing(models.Model):
+    _inherit = "mail.mass_mailing"
+
+    keep_archives = fields.Boolean('Keep Archives')
```
Comment (hbrunn) [inline mass_mailing_keep_archives/mass_mailing.py]: remove string parameter: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#field

**Comment by nicomacr** [comment]
thanks for the suggestions

**Comment by hbrunn** [comment]
:+1:

**Comment by atchuthan** [comment]
code reviewed :+1:

File: mass_mailing_keep_archives/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# © 2015 ADHOC SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+##############################################################################
+
+{
+    'name': 'Mass Mailing Keep Archives',
+    'version': '8.0.0.0.0',
+    'author': 'ADHOC SA',
```
Comment (hbrunn) [inline mass_mailing_keep_archives/__openerp__.py]: add OCA as author: https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py

**Pull request merged**